src/views/DashboardView/StudentDashboard/SubjectSelection/SubjectCard.vue

```diff
@@ -67,9 +67,17 @@ export default {
   components: { DropdownList, HyperlinkButton, LargeButton },
   data() {
     return {
-      selectedSubtopic: ""
+      selectedSubtopic: "",
+      timeoutId: null,
+      disableButton: false
     };
   },
+  mounted() {
+    this.checkOrEnforceWaitingPeriod();
```

Comment:
@jkeat ,

I spent way more time than I should have deciding on a name for this function, lol